pkg/ntpmonitor/ntpmonitor.go

```diff
@@ -44,10 +44,30 @@ var (
 	ErrDeltaTooSmall = errors.New("delta is too small")
 )
 
+type serverResponses struct {
+	tooFewServerResponses   bool
+	tooManyInvalidResponses bool
+}
+
+type NTPClient interface {
+	QueryWithOptions(srv string, opts ntp.QueryOptions) (*ntp.Response, error)
+}
+
+type LiveNTPClient struct{}
+
+func (c LiveNTPClient) QueryWithOptions(srv string, opts ntp.QueryOptions) (*ntp.Response, error) {
+	return ntp.QueryWithOptions(srv, opts)
+}
+
 // NTPMonitor compares the local time with a set of trusted NTP servers.
 type NTPMonitor struct {
-	cfg *Config
-	run atomic.Bool
+	cfg       *Config
+	run       atomic.Bool
+	ntpClient NTPClient
+}
+
+func (n *NTPMonitor) WithCustomClient(client NTPClient) {
```

Comment:
I'd propose making a `NewFromConfigWithClient` method instead of `WithCustomClient`. Otherwise, you're initializing a client twice, once in `NewFromConfig` and once in `WithCustomClient`.